tests/ops/arithmetic/test_adjoint_op.py

```diff
@@ -0,0 +1,437 @@
+
+    @pytest.mark.parametrize(
+        "base",
+        (qml.RX(1.23, wires=0), qml.Rot(1.23, 2.345, 3.456, wires=0), qml.CRX(1.234, wires=(0, 1))),
+    )
+    def test_parameter_frequencies(self, base):
+        assert Adjoint(base).parameter_frequencies == base.parameter_frequencies
+
+
+class TestQueueing:
+    """Test that Adjoint operators queue and update base metadata"""
+
+    def test_queueing(self):
+        """Test queuing and metadata when both Adjoint and base defined inside a recording context."""
+
+        with qml.tape.QuantumTape() as tape:
+            base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+            op = Adjoint(base)
+
+        assert tape._queue[base]["owner"] is op
+        assert tape._queue[op]["owns"] is base
+        assert tape.operations == [op]
+
+    def test_queueing_base_defined_outside(self):
+        """Test that base is added to queue even if it's defined outside the recording context."""
+
+        base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+        with qml.tape.QuantumTape() as tape:
+            op = Adjoint(base)
+
+        assert tape._queue[base]["owner"] is op
+        assert tape._queue[op]["owns"] is base
+        assert tape.operations == [op]
+
+
+class TestMatrix:
+    """Test the matrix method for a variety of interfaces."""
+
+    def check_matrix(self, x, interface):
+        """Compares matrices in a interface independent manner."""
+        base = qml.RX(x, wires=0)
+        base_matrix = base.matrix()
+        expected = qml.math.conj(qml.math.transpose(base_matrix))
+
+        mat = Adjoint(base).matrix()
+
+        assert qml.math.allclose(expected, mat)
+        assert qml.math.get_interface(mat) == interface
+
+    def test_matrix_autograd(self):
+        """Test the matrix of an Adjoint operator with an autograd parameter."""
+        self.check_matrix(np.array(1.2345), "autograd")
+
+    def test_matrix_jax(self):
+        """Test the matrix of an adjoint operator with a jax parameter."""
+        jnp = pytest.importorskip("jax.numpy")
+        self.check_matrix(jnp.array(1.2345), "jax")
+
+    def test_matrix_torch(self):
+        """Test the matrix of an adjoint oeprator with a torch parameter."""
+        torch = pytest.importorskip("torch")
+        self.check_matrix(torch.tensor(1.2345), "torch")
+
+    def test_matrix_tf(self):
+        """Test the matrix of an adjoint opreator with a tensorflow parameter."""
+        tf = pytest.importorskip("tensorflow")
+        self.check_matrix(tf.Variable(1.2345), "tensorflow")
+
+    def test_no_matrix_defined(self):
+        """Test that if the base has no matrix defined, then Adjoint.matrix also raises a MatrixUndefinedError."""
+        rng = np.random.default_rng(seed=42)
+        shape = qml.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
+        params = rng.random(shape)
+
+        base = qml.StronglyEntanglingLayers(params, wires=[0, 1])
```

Comment:
I would not make this test depend on the fact that this template has no matrix defined. We want to add matrix method to all operators in future, and then the poor dev has to go back and change this test which they didn't write...